pandas/io/tests/test_parsers.py

```diff
@@ -4128,6 +4130,34 @@ def test_bool_header_arg(self):
             with tm.assertRaises(TypeError):
                 pd.read_fwf(StringIO(data), header=arg)
 
+    def test_multithread_stringio_read_csv(self):
+        max_row_range = 10000
+        num_files = 100
+
+        bytes = ['\n'.join(['%d,%d,%d' % (i, i, i) for i in range(max_row_range)]).encode()
+                 for j in range(num_files)]
+        files = [BytesIO(b) for b in bytes]
+
+        # Read all files in many threads
+        pool = ThreadPool(8)
+        results = pool.map(pd.read_csv, files)
+        first_result = results[0]
+
+        for result in results:
+            tm.assert_frame_equal(first_result, result)
+
+    def test_multithread_path_read_csv(self):
+        df = DataFrame(np.random.rand(5000, 20), columns=map(str, xrange(20)))
+        num_files = 20
+        with tm.ensure_clean('__passing_str_as_dtype__.csv') as path:
+            df.to_csv(path, index=False)
+            pool = ThreadPool(4)
+            read_dataframes = pool.map(pd.read_csv, [path]*num_files)
+
+            for single_dataframe in read_dataframes:
+                tm.assert_frame_equal(df, single_dataframe, check_names=False)
+
+
 class TestMiscellaneous(tm.TestCase):
```

Comment:
sort of. I think this functionailty will be used to read a single file in parts (not the same file several times)

something like this:

```
import pandas as pd
from pandas.util import testing as tm
import numpy as np
from multiprocessing.pool import ThreadPool
pd.options.display.width=200
fn = 'test.csv'
np.random.seed(1234)

N = 1000000
df = pd.DataFrame(np.random.randn(N,5),columns=list('abcde'))
df['foo'] = 'foo'
df['bar'] = 'bar'
df['baz'] = 'baz'
df['date'] = pd.date_range('20000101 09:00:00',periods=N,freq='s')
df['int'] = np.arange(N)

df.to_csv(fn)

def f():
    return pd.read_csv(fn, index_col=0, header=0)

def g(procs=4):

    def reader(arg):
        start, nrows = arg

        # read from the start line for nrows
        if not start:
            return pd.read_csv(fn, index_col=0, header=0, nrows=nrows)

        # skip the header entirely
        return pd.read_csv(fn, index_col=0, header=None, skiprows=int(start)+1,nrows=nrows)

    # our tasks
    tasks = [ (N*i/procs, N/procs) for i in range(procs) ]
    pool = ThreadPool(processes=procs)

    # execute
    results = pool.map(reader, tasks)

    # collate the results
    # use the header from the first piece
    header = results[0].columns
    for r in results[1:]:
        r.columns = header
    return pd.concat(results)

def compare(result, expected):
    tm.assert_frame_equal(result, expected)
```